api/models/User.js

```diff
@@ -133,10 +133,32 @@ var UserSchema = new Schema({
   timeStamp : {type: Date, default: Date.now, required: true },
 });
 
+UserSchema.pre('validate', function(next) {
```

Comment:
If the user is not a facebook user, make sure they are not missing the necessary fields. Otherwise, carry on